source/common/network/connection_impl.cc

```diff
@@ -494,7 +443,7 @@ void ConnectionImpl::onWriteReady() {
     if (error == 0) {
       ENVOY_CONN_LOG(debug, "connected", *this);
       state_ &= ~InternalState::Connecting;
-      onConnected();
+      transport_socket_->onConnected();
```

Comment:
This seems a little weird - the Network::Connection tells the socket it's connected, and then the socket calls raiseEvents back on the connection.

I suspect there's a good reason to pass through the socket but I'm missing it :-)

Reply:
The main purpose of this is for SSL handshake
- Connection calls TransportSocket::onConnected then transport socket starts the handshake.
- transport socket call raiseEvents when handshake is complete.

Reply:
Cool, that's a level of detail which might be worth documenting somewhere (the API file?)  that onConnected is called on the transport when the underlying transport is connected, and raiseEvent(connected) is called when any transport-level handshakes are completed.

Reply:
Done